File: libembosser-nippon-driver/src/main/java/org/brailleblaster/libembosser/drivers/nippon/NipponEmbosser.java
package org.brailleblaster.libembosser.drivers.nippon;

import java.io.InputStream;
import java.util.Optional;
import java.util.stream.Stream;

import javax.print.PrintService;
import javax.print.StreamPrintServiceFactory;

import org.brailleblaster.libembosser.spi.EmbossException;
import org.brailleblaster.libembosser.spi.Embosser;
import org.brailleblaster.libembosser.spi.EmbossingAttributeSet;
import org.brailleblaster.libembosser.spi.Notification;
import org.brailleblaster.libembosser.spi.Rectangle;
import org.w3c.dom.Document;

public class NipponEmbosser implements Embosser {
	private final String id;
	private final String manufacturer;
	private final String model;
	private final Rectangle minimumPaper;
	private final Rectangle maximumPaper;
	public NipponEmbosser(String id, String model, Rectangle maximumPaper, Rectangle minimumPaper) {
		this.id = id;
		this.manufacturer = "Nippon";
		this.model = model;
		this.maximumPaper = maximumPaper;
		this.minimumPaper = minimumPaper;
	}

	@Override
	public String getId() {
		return id;
	}

	@Override
	public String getManufacturer() {
		return manufacturer;
	}

	@Override
	public String getModel() {
		return model;
	}

	@Override
	public void embossPef(PrintService embosserDevice, Document pef, EmbossingAttributeSet attributes)
			throws EmbossException {
		// TODO Auto-generated method stub

	}

	@Override
	public void embossBrf(PrintService embosserDevice, InputStream brf, EmbossingAttributeSet attributes)
			throws EmbossException {
		// TODO Auto-generated method stub

	}

	@Override
	public Rectangle getMaximumPaper() {
		return maximumPaper;
	}

	@Override
	public Rectangle getMinimumPaper() {
		return minimumPaper;
	}

	@Override
	public boolean supportsInterpoint() {
		// TODO Auto-generated method stub
		return false;
	}

	@Override
	public Optional<StreamPrintServiceFactory> getStreamPrintServiceFactory() {
		// TODO Auto-generated method stub
		return Optional.empty();
	}

	@Override
	public Stream<Notification> checkPrerequisites() {
		// TODO Auto-generated method stub
		return Stream.empty();
	}

}

File: libembosser-nippon-driver/src/main/kotlin/org/brailleblaster/libembosser/drivers/nippon/NipponEmbosser.kt
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * Copyright (c) 2023 American Printing House for the Blind
 */

package org.brailleblaster.libembosser.drivers.nippon

import org.brailleblaster.libembosser.spi.Embosser
import org.brailleblaster.libembosser.spi.EmbossException
import javax.print.PrintService
import org.brailleblaster.libembosser.spi.EmbossingAttributeSet
import org.brailleblaster.libembosser.spi.Rectangle
import org.w3c.dom.Document
import java.io.InputStream
import java.util.*
import javax.print.StreamPrintServiceFactory

class NipponEmbosser(
    private val id: String,
    private val model: String,
    private val maximumPaper: Rectangle,
    private val minimumPaper: Rectangle
) : Embosser {
    private val manufacturer = "Nippon"
    override fun getId(): String {
        return id
    }

    override fun getManufacturer(): String {
        return manufacturer
    }

    override fun getModel(): String {
        return model
    }

    @Throws(EmbossException::class)
    override fun embossPef(embosserDevice: PrintService, pef: Document, attributes: EmbossingAttributeSet) {
        // TODO Auto-generated method stub
    }

    @Throws(EmbossException::class)
    override fun embossBrf(embosserDevice: PrintService, brf: InputStream, attributes: EmbossingAttributeSet) {
        // TODO Auto-generated method stub
    }

    override fun getMaximumPaper(): Rectangle {
        return maximumPaper
    }

    override fun getMinimumPaper(): Rectangle {
        return minimumPaper
    }

    override fun supportsInterpoint(): Boolean {
        // TODO Auto-generated method stub
        return false
    }

    override fun getStreamPrintServiceFactory(): Optional<StreamPrintServiceFactory> {
        // TODO Auto-generated method stub
        return Optional.empty()
    }
}